Add project root and repository to project:open telemetry

dividedmind opened the pull request:
Fixes #629

dustinbyrne commented on src/lib/git.ts:
```diff
@@ -0,0 +1,24 @@
+  if (remotes.length === 0) return;
+
+  const origin = remotes.find(({ name }) => name === 'origin');
+  if (origin?.fetchUrl) return origin.fetchUrl;
+
+  const first = remotes.find(({ fetchUrl }) => !!fetchUrl);
+  if (!first?.fetchUrl) return;
+
+  return first.fetchUrl;
```
It's possible that `fetchUrl` contains basic auth credentials or access tokens in the URL. We should consider the following cases and strip out any potentially sensitive information:
```
https://$ACCESS_TOKEN@github.com/$ORG/$REPO
https://$USER:$PASSWORD@github.com/$ORG/$REPO
```

dividedmind replied:
Great point!

dustinbyrne reviewed (changes requested):
One small request regarding credentials in the git remote URL, but otherwise it's looking good!

dustinbyrne commented:
I rebased this before merging but [this test](https://github.com/getappmap/vscode-appland/blob/master/test/integration/authentication/appmapServerLogin.test.ts#L30) appears to be failing after the rebase with `master`.

dividedmind commented:
> I rebased this before merging but [this test](https://github.com/getappmap/vscode-appland/blob/master/test/integration/authentication/appmapServerLogin.test.ts#L30) appears to be failing after the rebase with `master`.

The failure appears to be unrelated, but I fixed it anyway.

appland-release commented:
:tada: This PR is included in version 0.69.0 :tada:

The release is available on:
- `v0.69.0`
- [GitHub release](https://github.com/getappmap/vscode-appland/releases/tag/v0.69.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
